events/s3_object_lambda.go

```diff
@@ -0,0 +1,64 @@
+package events
+
+type S3ObjectLambdaEvent struct {
+	XAmzRequestID        string                              `json:"xAmzRequestId"`
+	GetObjectContext     *S3ObjectLambdaGetObjectContext     `json:"getObjectContext,omitempty"`
+	ListObjectsContext   *S3ObjectLambdaListObjectsContext   `json:"listObjectsContext,omitempty"`
+	ListObjectsV2Context *S3ObjectLambdaListObjectsV2Context `json:"listObjectsV2Context,omitempty"`
+	HeadObjectContext    *S3ObjectLambdaHeadObjectContext    `json:"headObjectContext,omitempty"`
```

Comment:
Looking for another reference for these fields. The documentation I found so far only shows `getObjectContext`

https://docs.aws.amazon.com/AmazonS3/latest/userguide/olap-event-context.html

Reply:
PR for the rust event https://github.com/calavera/aws-lambda-events/pull/127 didn't contain additional docs for List/Head object.

Guess I gotta check this by hand, or ask the service team for a pointer.

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/olap-writing-lambda.html#olap-getobject-response

I followed these examples in the documentation.

Reply:
My bad! I forgot that I referenced that exact documentation when reviewing one of your previous commits! https://github.com/aws/aws-lambda-go/pull/536#discussion_r1434608488